install-deps.sh

```diff
@@ -21,16 +21,36 @@ if test $(id -u) != 0 ; then
 fi
 export LC_ALL=C # the following is vulnerable to i18n
 
-ARCH=`uname -m`
+ARCH=$(uname -m)
+
+function map_version_id_to_codename_debian {
+    case $VERSION_ID in
+        6) CODENAME=squeeze ;;
+        7) CODENAME=wheezy ;;
+        8) CODENAME=jessie ;;
+        9) CODENAME=stretch ;;
+        *) CODENAME=unsupported ;;
+    esac
+}
 
-if [ -n "$WITH_SEASTAR" ]; then
-    with_seastar=true
-else
-    with_seastar=false
-fi
+function map_version_id_to_codename_ubuntu {
+    case $VERSION_ID in
+        12.04) CODENAME=precise ;;
+        16.04) CODENAME=xenial ;;
+        18.04) CODENAME=bionic ;;
+        *) CODENAME=unsupported ;;
+    esac
+}
+
+function map_version_id_to_codename {
```

Comment:
nit, instead of setting a global variable, i'd suggest `echo` the codename and set a local variable.

in this script, variable names with capital characters like `FOO_BAR` are used for imported variable and global ones. if `CODENAME` is not used everywhere, might want to use something like `codename` instead.

also, might want to squeeze these three functions into a single one.